packages/serverless-orchestration/src/ServerlessHub.js

```diff
@@ -99,50 +101,57 @@ hub.post("/", async (req, res) => {
       results
     });
 
-    // Validate that the promises returned correctly. If ANY have error, then catch them and throw them all together.
+    // Validate that the promises returned correctly. If any spokes rejected it is possible that it was due to a networking
+    // or internal GCP error. Re-try these executions. If a response is code 500 or contains an error then log it as an error.
     let errorOutputs = {};
     let validOutputs = {};
+    let retriedOutputs = [];
     results.forEach((result, index) => {
-      // If the child process in the spoke crashed it will return 500 (rejected). OR If the child process exited
-      // correctly but contained an error.
-      if (
-        result.status == "rejected" ||
-        (result.value && result.value.execResponse && result.value.execResponse.error) ||
-        (result.reason && result.reason.code == "500")
-      ) {
-        errorOutputs[Object.keys(configObject)[index]] = {
-          status: result.status,
-          execResponse:
-            (result.value && result.value.execResponse) ||
-            (result.reason &&
-              result.reason.response &&
-              result.reason.response.data &&
-              result.reason.response.data.execResponse),
-          botIdentifier: Object.keys(configObject)[index]
-        };
-      } else {
-        validOutputs[Object.keys(configObject)[index]] = {
-          status: result.status,
-          execResponse: result.value && result.value.execResponse,
-          botIdentifier: Object.keys(configObject)[index]
-        };
+      if (result.status == "rejected") {
```

Comment:
So IIUC this is the main addition, where you catch rejected spoke outputs and push them into an array to retry?

Reply:
That's right. if the spoke call is rejected then we generate a separate array that is executed below. It's only the rejected execution types that have this re-try logic. Other errors are caught as before and logged.